### ISSUE TITLE
Adding user to branch permission whitelist fails

### ISSUE THREAD

**Issue by cardonator** [gogits/gogs#4333]
- Gogs version (or commit ref): 0.10.26.0323
- Git version: 2.7.4
- Operating system: Ubuntu 16.04.2
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist (usually found in `log/gogs.log`):
```
2017/03/23 22:50:47 [ERROR] [...s/context/context.go:105 Handle()] UpdateOrgProtectBranch/UpdateProtectBranch: insert new protect branch whitelists: UNIQUE constraint failed: protect_branch_whitelist.repo_id, protect_branch_whitelist.name, protect_branch_whitelist.user_id
2017/03/23 22:50:47 [TRACE] Template: status/500
```

## Description
Attempting to add users to the whitelist of users that can push to a branch when branch protection is on leads to an error. I pasted this error above in the log section. That's the only relevant section of the log. Everything around it is CSRF tokens and session IDs.

Basically create a repo, turn on branch protection for master and add a user to the whitelist. Then try to add a second user. I was able to replicate this on two repos, but I couldn't access this feature on try.gogs.io to see if I could reproduce it there.

**Comment by Unknwon** [comment]
>but I couldn't access this feature on try.gogs.io

Why can't?

**Comment by cardonator** [comment]
It was greyed out on the account/repo I created for some reason.

**Comment by Unknwon** [comment]
Please attach screenshot.

**Comment by cardonator** [comment]
I realized I hadn't pushed anything to the repo I created. I see the branch permissions for master now, but I can't see the whitelist still. I do have two collaborators on the repo.

![image](https://cloud.githubusercontent.com/assets/7117256/24309413/527f13a4-1091-11e7-9173-d33f10d45b25.png)

**Comment by Unknwon** [comment]
Only organizational repository can have protect branch whitelist.

**Comment by cardonator** [comment]
Ah! Okay, I'm testing that now.

**Comment by cardonator** [comment]
Okay, I was able to replicate it on my test repo inside an org repo.

https://try.gogs.io/cardonatororg/test-repos

**Comment by Unknwon** [comment]
>a second person

Which user?

**Comment by Unknwon** [comment]
What your reproduce steps to cause this 500?

Because I can't reproduce myself and only found bad data on your test repository.

**Comment by cardonator** [comment]
It's basically:

* Create repo in org
* Add collaborator to repo
* turn on branch permissions
* turn on whitelist
* add one user
* Save
* try to add second user
* Save
* 500

**Comment by Unknwon** [comment]
Thank you!

Patch has pushed to fix this issue, please test on `develop` branch or https://try.gogs.io.

**Comment by cardonator** [comment]
Appears to work on the demo site, thanks!!
